### PR TITLE
Miss text changing

### PR DESCRIPTION
**Pull request by Shiftenas**
There are the (I hope) true version of my pull request

### CODE REVIEW

File: Documentation CYF 1.0/pages/variables.html [hunk 1]
```diff
@@ -239,6 +239,8 @@ <h4>Monster script variables</h4>
                 This overrides the random dialogue and is meant for special actions (e.g. you hit Vegetoid's green carrots after selecting Dinner from the 
                 ACT menu). This variable gets <i>cleared every time after it's read out in the monster dialogue phase</i>. This is done
                 so you don't have to take care of managing it manually.</p>
+                <p><span class="CYF"></span> <span class="string"></span> <span class="term">defensemisstext</span> - The text who will be displayed when the <span class="term">def</span> is too high. "MISS" by default</p>
```
Comment (RhenaudTheLukark) [inline Documentation CYF 1.0/pages/variables.html]: The text which will be displayed if the Player's attack is successful but deals 0 damage. "MISS" by default.

File: Assets/Scripts/Battle/FightUIController.cs [hunk 2]
```diff
@@ -249,8 +249,10 @@ public class FightUIController : MonoBehaviour {
                 for (int i = 0; i < boundFightUiInstances.Count; i++) {
                     boundFightUiInstances[i].enemy.HandleAttack(-1);
                     StationaryMissScript smc2 = Instantiate(smc);
+                    if (boundFightUiInstances[i].enemy.NoAttackMissText != null) smc2.SetText(boundFightUiInstances[i].enemy.NoAttackMissText);
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Battle/FightUIController.cs]: We can add a new line after the condition, this line's awfully long.

File: Assets/Scripts/Battle/LuaEnemyController.cs [hunk 3]
```diff
@@ -128,6 +128,14 @@ public class LuaEnemyController : EnemyController {
         get { return (int)script.GetVar("gold").Number; }
         set { script.SetVar("gold", DynValue.NewNumber(value)); }
     }
+    public string DefenseMissText {
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Battle/LuaEnemyController.cs]: Don't forget to add an empty line between each declaration.

File: Assets/Scripts/Battle/StationaryMissScript.cs [hunk 4]
```diff
@@ -6,15 +6,20 @@
...
 
+    public void SetText(string _text) {
+        text = _text;
+    }
+
 	void Start () {
         TextManager mgr = GetComponent<TextManager>();
         mgr.SetFont(SpriteFontRegistry.Get(SpriteFontRegistry.UI_DAMAGETEXT_NAME));
-        mgr.SetText(new TextMessage("[color:c0c0c0]MISS", false, true));
+        mgr.SetText(new TextMessage("[color:c0c0c0]" + text.ToUpper(), false, true));
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Battle/StationaryMissScript.cs]: I think people should be able to use lowercase characters, the font handles them, after all.

**Comment by Shiftenas** [comment]
The changes are done

File: Assets/Scripts/Battle/FightUIController.cs [hunk 5]
```diff
@@ -249,8 +249,12 @@ public class FightUIController : MonoBehaviour {
                 for (int i = 0; i < boundFightUiInstances.Count; i++) {
                     boundFightUiInstances[i].enemy.HandleAttack(-1);
                     StationaryMissScript smc2 = Instantiate(smc);
+                    if (boundFightUiInstances[i].enemy.NoAttackMissText != null)
+                    smc2.SetText(boundFightUiInstances[i].enemy.NoAttackMissText);
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Battle/FightUIController.cs]: Indenting!!!

File: Assets/Scripts/Battle/FightUIController.cs [hunk 6]
```diff
@@ -249,8 +249,12 @@ public class FightUIController : MonoBehaviour {
...
                     smc2.transform.SetParent(GameObject.Find("Canvas").transform);
-                    smc2.setXPosition(boundFightUiInstances[i].enePos.x);
+                    if (boundFightUiInstances[i].enemy.NoAttackMissText != null)
+                    smc2.setXPosition(boundFightUiInstances[i].enePos.x - 10 * boundFightUiInstances[i].enemy.NoAttackMissText.Length + 20);
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Battle/FightUIController.cs]: Indenting!!!

File: Documentation CYF 1.0/pages/api-functions-misc.html [hunk 7]
```diff
@@ -199,8 +199,7 @@ <h3>File Functions</h3>
                 <li><span class="term">File.Delete()</span> <span class="ref">(w)</span>:
                     Deletes the file.<br>
                     If using <span class="term">"w"</span> or <span class="term">"rw"</span> mode, you can use <span class="term">File.Write</span> to create the file again.</li><br>
-                <li><span class="term">File.Move(<span class="string"></span> new_path)</span>:
-                    Moves the file to the new_path, can be used to rename a file.</li><br>
```
Comment (RhenaudTheLukark) [inline Documentation CYF 1.0/pages/api-functions-misc.html]: You forgot this line...

File: Documentation CYF 1.0/pages/variables.html [hunk 8]
```diff
@@ -257,8 +259,6 @@ <h4>Monster script variables</h4>
                 change this at any time, but this <i>must</i> be initially set to something. For a list of all possible options, check the
                 <a href="../media/dialogoptions.png">dialog bubble names</a> chart; it's also in the sidebar.
                 Positioning of the bubbles is done automatically.</p>
-            <p><span class="CYF"></span> <span class="string"></span> <span class="term">dialogueprefix</span> - A string, appended to the beginning
```
[the remark this answers is not in the document]
Comment (RhenaudTheLukark) [reply Documentation CYF 1.0/pages/variables.html]: Hasn't been resolved in the last commit.

**Comment by Shiftenas** [comment]
This time, I've repaired everything

**Comment by WD200019** [comment]
I saw on the Discord that the text might not be centered correctly on the enemy, do you think you could double check to make sure both variables still keep the text centered? Other than that I think this looks nice

**Comment by Shiftenas** [comment]
I've already worked on that

File: Documentation CYF 1.0/pages/api-functions-misc.html [hunk 9]
```diff
@@ -199,8 +199,8 @@ <h3>File Functions</h3>
...
+                    <li><span class="term">File.Move(<span class="string"></span> new_path)</span>:
```
Comment (RhenaudTheLukark) [inline Documentation CYF 1.0/pages/api-functions-misc.html]: Last thing, you added spaces that weren't there before!

File: Documentation CYF 1.0/pages/api-functions-misc.html [hunk 10]
```diff
@@ -199,8 +199,8 @@ <h3>File Functions</h3>
...
+                        Moves the file to the new_path, can be used to rename a file.</li><br>
```
Comment (RhenaudTheLukark) [inline Documentation CYF 1.0/pages/api-functions-misc.html]: Here too!

**Comment by RhenaudTheLukark** [comment]
We're almost there! Courage!

**Comment by Shiftenas** [comment]
Repaired

**Comment by RhenaudTheLukark** [comment]
The pull request has been cancelled. You requested it on the master branch, even though I specifically told you to do it on the feature branch.

**Pull request merged**